tests/timeseries/_battery_pool/test_battery_pool_control_methods.py

```diff
@@ -0,0 +1,370 @@
+                    battery_id,
+                    now,
+                    soc=50.0,
+                    soc_lower_bound=10.0,
+                    soc_upper_bound=90.0,
+                    power_exclusion_lower_bound=0.0,
+                    power_exclusion_upper_bound=0.0,
+                    power_inclusion_lower_bound=-1000.0,
+                    power_inclusion_upper_bound=1000.0,
+                    capacity=2000.0,
+                ),
+                0.05,
+            )
+
+    async def _init_data_for_inverters(self, mocks: Mocks) -> None:
+        now = datetime.now(tz=timezone.utc)
+        for inv_id in mocks.microgrid.battery_inverter_ids:
+            mocks.streamer.start_streaming(
+                InverterDataWrapper(
+                    inv_id,
+                    now,
+                    active_power_exclusion_lower_bound=0.0,
+                    active_power_exclusion_upper_bound=0.0,
+                    active_power_inclusion_lower_bound=-1000.0,
+                    active_power_inclusion_upper_bound=1000.0,
+                ),
+                0.05,
+            )
+
+    def _assert_report(
+        self,
+        report: _power_managing.Report,
+        *,
+        power: float | None,
+        lower: float,
+        upper: float,
+        expected_result_pred: typing.Callable[[power_distributing.Result], bool]
+        | None = None,
+    ) -> None:
+        assert report.target_power == (
+            Power.from_watts(power) if power is not None else None
+        )
+        assert (
+            report.inclusion_bounds is not None and report.exclusion_bounds is not None
+        )
+        assert report.inclusion_bounds.lower == Power.from_watts(lower)
+        assert report.inclusion_bounds.upper == Power.from_watts(upper)
+        assert report.exclusion_bounds.lower == Power.from_watts(0.0)
+        assert report.exclusion_bounds.upper == Power.from_watts(0.0)
+        if expected_result_pred is not None:
+            assert report.distribution_result is not None
+            assert expected_result_pred(report.distribution_result)
+
+    async def test_case_1(
+        self,
+        mocks: Mocks,
+        mocker: MockerFixture,
+    ) -> None:
+        """Test case 1.
+
+        - single battery pool with all batteries.
+        - all batteries are working, then one battery stops working.
+        """
+        set_power = typing.cast(
+            AsyncMock, microgrid.connection_manager.get().api_client.set_power
+        )
+
+        await self._patch_battery_pool_status(mocks, mocker)
+        await self._init_data_for_batteries(mocks)
+        await self._init_data_for_inverters(mocks)
+
+        battery_pool = microgrid.battery_pool()
+
+        # This is used just to wait for the processing to be complete.  The results are
+        # not used.
+        #
+        # It will be replaced by a reporting streaming from the PowerManager in a
+        # subsequent commit.
```

Comment:
Is this something you meant to update in this PR?

Reply:
oh yes, that whole comment is no longer relevant.  Will remove.

Reply:
done